src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -49,102 +49,90 @@ protected void collectInputs() {}
-          setStateFromRequest(RobotState.PASS_SHOOTING);
-        }
-      }
-      case UNJAM -> {}
-      case INTAKING -> {
-        if (intake.hasNote() || queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_WITH_GP);
-        }
-      }
-      case OUTTAKING -> {
-        if (!intake.hasNote() && !queuer.hasNote()) {
-          setStateFromRequest(RobotState.IDLE_NO_GP);
-        }
-      }
-    }
+      case SPEAKER_PREPARE_TO_SCORE ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.SPEAKER_SCORING : currentState;
+
+      case AMP_PREPARE_TO_SCORE ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.AMP_SCORING : currentState;
 
-    return currentState;
+      case FEEDING_PREPARE_TO_SHOOT ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.FEEDING_SHOOTING : currentState;
+      case PASS_PREPARE_TO_SHOOT ->
+          shooter.atGoal() && arm.atGoal() ? RobotState.PASS_SHOOTING : currentState;
+      case UNJAM -> currentState;
+      case INTAKING -> queuer.hasNote() ? RobotState.IDLE_WITH_GP : currentState;
+      case OUTTAKING ->
+          !queuer.hasNote() || !intake.hasNote() ? RobotState.IDLE_NO_GP : currentState;
+    };
   }
 
   @Override
   protected void afterTransition(RobotState newState) {
     switch (newState) {
-      case SPEAKER_PREPARE_TO_SCORE, SPEAKER_SCORING, SPEAKER_WAITING -> {
+      case SPEAKER_PREPARE_TO_SCORE, SPEAKER_WAITING -> {
         arm.setState(ArmState.SPEAKER_SHOT);
         shooter.setState(ShooterState.SPEAKER_SHOT);
         intake.setState(IntakeState.IDLE);
         queuer.seState(QueuerState.IDLE_WITH_GP);
       }
-      case AMP_PREPARE_TO_SCORE, AMP_SCORING, AMP_WAITING -> {
+      case SPEAKER_SCORING -> {
+        arm.setState(ArmState.SPEAKER_SHOT);
+        shooter.setState(ShooterState.SPEAKER_SHOT);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.SHOOTING);
+      }
+      case AMP_PREPARE_TO_SCORE, AMP_WAITING -> {
         arm.setState(ArmState.AMP);
         shooter.setState(ShooterState.AMP);
         intake.setState(IntakeState.IDLE);
         queuer.seState(QueuerState.IDLE_WITH_GP);
       }
-      case FEEDING_PREPARE_TO_SHOOT, FEEDING_SHOOTING, FEEDING_WAITING -> {
+      case AMP_SCORING -> {
+        arm.setState(ArmState.AMP);
+        shooter.setState(ShooterState.AMP);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.SHOOTING);
+      }
+      case FEEDING_PREPARE_TO_SHOOT, FEEDING_WAITING -> {
         arm.setState(ArmState.FEEDING);
         shooter.setState(ShooterState.FEEDING);
         intake.setState(IntakeState.IDLE);
         queuer.seState(QueuerState.IDLE_WITH_GP);
       }
-      case PASS_PREPARE_TO_SHOOT, PASS_SHOOTING, PASS_WAITING -> {
+      case FEEDING_SHOOTING -> {
+        arm.setState(ArmState.FEEDING);
+        shooter.setState(ShooterState.FEEDING);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.SHOOTING);
```

Comment:
just realized there's a typo in `QueuerSubsystem` itself, `seState` should be `setState`